indico/modules/categories/client/js/index.js

```diff
@@ -5,64 +5,33 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
-import './display';
+import categoryStatisticsURL from 'indico-url:categories.statistics';
 
-(function(global) {
-  global.setupCategoryStats = function setupCategoryStats() {
-    $(document).ready(() => {
-      $('.plot-container .plot').each((__, element) => {
-        const $this = $(element);
-        const xMin = $this.data('min-x');
-        const xMax = $this.data('max-x');
-        const yMin = $this.data('min-y');
-        const yMax = $this.data('max-y');
+import React from 'react';
+import ReactDOM from 'react-dom';
 
-        var data = $this.data('values') || {};
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
 
-        const currentYear = new Date().getFullYear();
-        const current = [[currentYear, data[currentYear]]];
+import './display';
+import CategoryStatistics from './components/CategoryStatistics';
 
-        const options = {
-          axes: {
-            xaxis: {
-              label: $this.data('label-xaxis'),
-              max: xMax,
-              min: xMin,
-              tickOptions: {showGridline: false},
-            },
-            yaxis: {
-              label: $this.data('label-yaxis'),
-              max: yMax,
-              min: yMin,
-            },
-          },
-          height: 400,
-          highlighter: {
-            location: 'n',
-            tooltipAxes: 'yx',
-            tooltipSeparator: $this.data('qbubble'),
-          },
-          series: [
-            {
-              fillAlpha: 0.9,
-              fill: true,
-              fillAndStroke: true,
-            },
-            {
-              markerOptions: {
-                color: '#005272',
-                style: 'circle',
-              },
-            },
-          ],
-          width: 400,
-        };
-        data = [_.pairs(data).map(datum => [parseInt(datum[0], 10), datum[1]]), current];
-        const plot = global.$.jqplot(element.id, data, global.processJqPlotOptions(options));
-        $(window).resize(() => {
-          plot.replot({resetAxes: true});
-        });
-      });
+(function(global) {
+  global.setupCategoryStats = function setupCategoryStats() {
+    document.addEventListener('DOMContentLoaded', async () => {
+      const rootElement = document.querySelector('#category-content-wrapper');
+      if (!rootElement) {
+        return;
+      }
+      const categoryId = parseInt(rootElement.dataset.categoryId, 10);
+      let response;
+      try {
+        response = await indicoAxios.get(categoryStatisticsURL({category_id: categoryId}));
+      } catch (e) {
+        handleAxiosError(e);
+        return;
+      }
+      const data = response.data;
+      ReactDOM.render(<CategoryStatistics categoryId={categoryId} data={data} />, rootElement);
```

Comment:
you aren't using the categoryId prop. what about moving the data loading into the statistics component though instead of doing it here?